mulog-cloudwatch/src/com/brunobonacci/mulog/publishers/cloudwatch.clj

```diff
@@ -0,0 +1,116 @@
+(ns com.brunobonacci.mulog.publishers.cloudwatch
+  (:require [com.brunobonacci.mulog.publisher :as p]
+            [com.brunobonacci.mulog.buffer :as rb]
+            [com.brunobonacci.mulog.utils :as ut]
+            [com.brunobonacci.mulog.common.json :as json]
+            [cognitect.aws.client.api :as aws]))
+
+(defn- has-invalid-token?
+  [type]
+  (= "InvalidSequenceTokenException" type))
+
+
+(defn- has-anomaly?
+  [rs]
+  (contains? rs :cognitect.anomalies/category))
+
+
+(defn- create-cloudwatch-client
+  [params]
+  (aws/client params))
+
+
+(defn- publish!
+  [cloudwatch-client group-name stream-name records next-token]
+  (let [rq {:logGroupName  group-name
+            :logStreamName stream-name
+            :logEvents     records}
+        token @next-token
+        rs  (aws/invoke cloudwatch-client {:op  :PutLogEvents
+                                           :request (if (nil? token)
+                                                     rq
+                                                     (merge rq token))})]
+
+    (if (has-anomaly? rs)
+      (if (has-invalid-token? (:__type rs) )
```

Comment:
maybe better if `has-invalid-token?` just takes the response `rs` so that the logic on what to check is entirely inside the function

Reply:
yeah, good idea, fixed